src/scripts/main.js

```diff
@@ -1,3 +1,258 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('.button');
+const startMessage = document.querySelector('.message_start');
+const scoreText = document.querySelector('.game_score');
+const table = document.querySelector('.game_field');
+const winMessage = document.querySelector('.message_win');
+const loseMessage = document.querySelector('.message_lose');
+
+let board;
+let score = 0;
+const rows = 4;
+const columns = 4;
+
+button.addEventListener('click', () => {
+  if (button.classList.contains('start')) {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.textContent = 'Restart';
+    startMessage.classList.add('hidden');
+  } else {
+    winMessage.classList.add('hidden');
+    loseMessage.classList.add('hidden');
+  };
+
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  score = 0;
+  scoreText.textContent = score;
+
+  setTwo();
+  setTwo();
+});
+
+document.addEventListener('keydown', (e) => {
+  switch (e.key) {
+    case 'ArrowLeft':
+      movedLeft();
+      break;
+
+    case 'ArrowRight':
+      movedRight();
+      break;
+
+    case 'ArrowUp':
+      movedUp();
+      break;
+
+    case 'ArrowDown':
+      movedDown();
+      break;
+  }
+
+  if (gameOver(board)) {
```

Comment:
It would be better to move functions call to variables to make code more readable